packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -83,24 +86,25 @@ export class CrossChainMessenger implements ICrossChainMessenger {
     this.l1SignerOrProvider = toSignerOrProvider(opts.l1SignerOrProvider)
     this.l2SignerOrProvider = toSignerOrProvider(opts.l2SignerOrProvider)
     this.l1ChainId = toNumber(opts.l1ChainId)
```

Comment:
Why require the chain id to be passed in when there is a provider passed in that could fetch it for you? It does simplify needing to fetch it in a function that may be sync